src/components/activity-groups-list/activity-group-item/activity-item/route3.tsx

```diff
@@ -0,0 +1,149 @@
+import { Route3Activity } from '@temple-wallet/transactions-parser';
+import { BigNumber } from 'bignumber.js';
+import { isEmpty } from 'lodash-es';
+import React, { FC, memo, useMemo } from 'react';
+import { Text, View } from 'react-native';
+
+import { Divider } from 'src/components/divider/divider';
+import { ExternalLinkButton } from 'src/components/icon/external-link-button/external-link-button';
+import { Icon } from 'src/components/icon/icon';
+import { IconNameEnum } from 'src/components/icon/icon-name.enum';
+import { PublicKeyHashText } from 'src/components/public-key-hash-text/public-key-hash-text';
+import { useRobotIconStyles } from 'src/components/robot-icon/robot-icon.styles';
+import { calculateFiatAmount, useNonZeroAmounts } from 'src/hooks/use-non-zero-amounts.hook';
+import { ActivityAmount } from 'src/interfaces/non-zero-amounts.interface';
+import { useUsdToTokenRates } from 'src/store/currency/currency-selectors';
+import { useFiatToUsdRateSelector, useSelectedRpcUrlSelector } from 'src/store/settings/settings-selectors';
+import { formatSize } from 'src/styles/format-size';
+import { TEMPLE_TOKEN } from 'src/token/data/tokens-metadata';
+import { toTokenSlug } from 'src/token/utils/token.utils';
+import { calculateDollarValue, separateAmountsBySign } from 'src/utils/activity.utils';
+import { isDefined } from 'src/utils/is-defined';
+import { tzktUrl } from 'src/utils/linking';
+import { mutezToTz } from 'src/utils/tezos.util';
+
+import { ActivityGroupAmountChange } from '../activity-group-amount-change/activity-group-amount-change';
+import { ActivityGroupDollarAmountChange } from '../activity-group-dollar-amount-change/activity-group-dollar-amount-change';
+import {
+  useActivityCommonStyles,
+  useActivityDetailsStyles,
+  useActivityGroupItemStyles
+} from '../activity-group-item.styles';
+import { ItemAmountChange } from '../item-amount-change/item-amount-change';
+import { ActivityGroupItemSelectors } from '../selectors';
+import { AbstractItem } from './abstract-item';
+
+export const Route3: FC<{ activity: Route3Activity }> = memo(({ activity }) => {
+  const nonZeroAmounts = useNonZeroAmounts(activity.tokensDeltas);
+
+  return (
+    <AbstractItem
+      status={activity.status}
+      timestamp={activity.timestamp}
+      face={<Face nonZeroAmounts={nonZeroAmounts} />}
+      details={<Details hash={activity.hash} nonZeroAmounts={nonZeroAmounts} cashback={activity.cashback} />}
+    />
+  );
+});
+
+const Face: FC<{ nonZeroAmounts: Array<ActivityAmount> }> = ({ nonZeroAmounts }) => {
```

Comment:
maybe move Face and Details to separate files?

Reply:
This Face and Details components is only used for this activity type, only in this file. And there were another activity items added this way, so i just followed the code style.